### PR TITLE
Capital indexed bond: trade model and pricer

### LINKED ISSUE #784: Inflation-indexed bond: product and trade

### LINKED ISSUE #785: Inflation-indexed bond: pricer

### PR DESCRIPTION
**Pull request by yukiiwashita**
* This work closes #784 
* This work closes #785

### CODE REVIEW

File: modules/product/src/main/java/com/opengamma/strata/product/bond/YieldConvention.java [hunk 1]
```diff
@@ -21,7 +21,8 @@
  * specific to each yield convention. Accordingly, the computation of price, convexity
  * and duration from the yield should be based on this yield convention. 
  * <p>
- * Reference: "Bond Pricing", OpenGamma Documentation 5, Version 2.0, May 2013.
+ * References: "Bond Pricing", OpenGamma Documentation 5, Version 2.0, May 2013, 
+ * "Inflation Instruments: Swap Zero-coupon, Year-on-year and Bonds."
  */
 public enum YieldConvention {
```
Comment (marc-opengamma) [inline modules/product/src/main/java/com/opengamma/strata/product/bond/YieldConvention.java]: Do we want a single YieldConvention with all the conventions: Fixed Coupon bonds, Inflation, Bills, etc. or does it make sense to have different enum?

File: modules/product/src/main/java/com/opengamma/strata/product/bond/CapitalIndexedBondTrade.java [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final SecurityLink<CapitalIndexedBond> securityLink;
+  /**
+   * The quantity, indicating the number of bond contracts in the trade.
+   * <p>
+   * This will be positive if buying and negative if selling.
+   */
+  @PropertyDefinition
+  private final long quantity;
+
```
Comment (marc-opengamma) [inline modules/product/src/main/java/com/opengamma/strata/product/bond/CapitalIndexedBondTrade.java]: Do we need a "payment" of some type to represent the settlement (price, date, etc.)?

File: modules/product/src/main/java/com/opengamma/strata/product/bond/YieldConvention.java (hunk 1 above)
Comment (jodastephen) [inline modules/product/src/main/java/com/opengamma/strata/product/bond/YieldConvention.java]: When I looked at OG-Platform Biill yield conventions had nothing in common with bond ones, so I think separate types makes sense.

More broadly, I don't like the `if(convention = FOO)` style, as really the logic should be in the convention, but that is a bigger change.

**Comment by yukiiwashita** [comment]
This work is closed by #835

**Pull request closed without merging**